src/compiler/transformers/es2015.ts

```diff
@@ -3338,7 +3338,7 @@ namespace ts {
         priority: 0,
         text: `
             var __extends = (this && this.__extends) || function (d, b) {
-                for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
+                Object.setPrototypeOf(d, b);

```

Comment:
I'm not familiar with the new transformer based emitter. But it seems `Object.setPrototypeOf` is not supported in ES3 environment and it is emitted in ES3 target. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf